Core/Src/main.c

```diff
@@ -87,14 +90,34 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); //CS default high
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1); //PWM on Channel 1
+
+  uint8_t txData[3] = {0x01, 0x80, 0x00};
+  uint8_t rxData[3] = {0};
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //CS low start communication
+	  HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 3, 1000);
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); //CS high end communication
+
+	  uint16_t adc_value = 0;
+	  adc_value = (rxData[1] << 8) | rxData[2]; //conversion of ADC value to counts
+
+	  uint16_t pwm_value = 0;
+	  pwm_value = ((float)adc_value/MAX_ADC_VALUE)*3000 + 3000;
```

Comment:
Correct formula, but this is not 100 percent, I need to know your timer period to verify the result is ultimately correct. But you don't have a spi.c file, I guess this is caused by the ioc issue you were talking about?

Reply:
I calculated the period to be 60 000